docs/BBA-EN.md

```diff
@@ -0,0 +1,68 @@
+# Binary Byzantine Agreement
+
+### Overview
+
+Basically Byzantine Agreement is an algorithm to have a common piece of information in spite of the fact that there might be failed or faulty process in the network.
+
+The correct processes communicate only through messages and can correctly identifies the  sender of each message. But processes are assumed to have no authentication, whether the sender Bob is really Bob, so there is no immediate way of detecting whether the message is corrupted. Failed process does not perform the actions prescribed by the Byzantine Agreement algorithm.
+
+The cost of extending a binary Byzantine Agreement algorithm for multivalue is very expensive when the transmission cost is high. The larger the size of the domain, the higher the cost. 
+
+And this [paper](<https://groups.csail.mit.edu/tds/papers/Coan/TurpinCoan-ipl84.pdf>) devises **two rounds mechanism** for distributed p2p network to reach consensus on a value of domain V, **which significantly reduced its transmission cost.**
+
+### Requirements
+
+The requirements for this algorithm to work is if `P` is the number of members that engaged in the agreement, **the number of failed member `T` must satisfies `P > 3T` condition.**
+
+### About Algorithm
+
+#### First round
+
+each member sends its initial value to every other members. And each member is said to be *perplexed* if, in the first round, it receives at least as many as `(1/2)*(P-T)` initial values different from its own. Members that are not perplexed are said to be *content*.
+
+#### Second round
+
+Each perplexed member sends a message to every other members, the meaning of this message is to notifiy "I am perplexed". Each member maintains three local variables: two arrays indexed by member id and a boolean. For member `j`, and for `i` which is  `i != j` , these variables are defined as follows:
+
+```
+v(j): the member's initial value
+v(i): the initial value received from member i
+
+p(j): a boolean value and is TRUE when this member is perplexed
+p(i): a boolean value and is TRUE when member i claiming it is perplexed
+
+alert: a boolean value and is TRUE when at least as many as (P-2T) elements of p are true. 
+```
+
+If `alert == true`, then it means that there are correct members with different initial values from V. In this case all correct member use a predefined default value from V.
```

Comment:
It's just question. 
1. As mentioned above, `V` is domain. What is domain? Please give an example

2. same as question 1. as mentioned this : `In this case all correct member use a predefined default value from V` what is predefined default value from V?

3. Is perplexed member and byzantine different?

4. In case of `alert ==true`, how to distinguish correct node? 

And I think... To satisfy the two expressions `P>3T`, `(1/2)(P-T)`, A perplexed member is found when P> 3T is not satisfied. I don't understand the condition of `alert==true` when `P-2T`

Reply:
1. domain `V` can be any thing. it can be candidate/leader for example, agreement is about electing leader among candidates. or transaction, agreement can be about consensus on transaction.

2. predefined value from V can be, for example, in the case of candidate, it can be old leader. In the case of transaction predefined value can be null

3. Yes different. perplexed member is correct process, byzantine is failed process who do not send messages.

4. There's no need to distinguish who is correct node. If `alert == true` then we pick the predefined value from V. And the proof is in the Proof section. (this will be talk about more deeply in Saturday)

5. If I understand the last question correctly, let's assume one situation, all members (correct or failed) received different initial value. In that case all the values they receive will different. And all will perplexed

Reply:
hmm.. I don't know well if i understand correctly. 
When P = 12 and T = 3 (3 byzantine nodes) then (1/2)(P-T) = 4.5 (4 or 5 when apply rounding)
then any node in network claim "i am perplexed" only when it finds 4 or 5 different initial value. but T = 3, so never exceed 4 or 5. 

Is it incorrect idea? or did i think too narrow? The above assumption is only when same epoch.